clients/pubsub.go

```diff
@@ -46,21 +46,51 @@ func NewClient(ctx context.Context, cfg config.Source) (*PubSub, error) {
 		return nil, fmt.Errorf("new pubsub client: %w", err)
 	}
 
+	cctx, cancel := context.WithCancel(ctx)
+
 	pubSub := &PubSub{
 		Cli:           cli,
-		MessagesCh:    make(chan *pubsub.Message, maxOutstandingMessages),
-		AckMessagesCh: make(chan *pubsub.Message, maxOutstandingMessages),
+		MessagesCh:    make(chan *pubsub.Message, pubsub.DefaultReceiveSettings.MaxOutstandingMessages),
+		AckMessagesCh: make(chan *pubsub.Message, pubsub.DefaultReceiveSettings.MaxOutstandingMessages),
 		ErrorCh:       make(chan error),
+		canceledCh:    make(chan struct{}),
+		ctxCancel:     cancel,
 	}
 
 	go func() {
-		err = pubSub.Cli.Subscription(cfg.SubscriptionID).Receive(ctx, func(ctx context.Context, m *pubsub.Message) {
+		err = pubSub.Cli.Subscription(cfg.SubscriptionID).Receive(cctx, func(ctx context.Context, m *pubsub.Message) {
 			pubSub.MessagesCh <- m
 		})
 		if err != nil {
 			pubSub.ErrorCh <- fmt.Errorf("subscription receive: %w", err)
 		}
+
+		pubSub.canceledCh <- struct{}{}
 	}()
 
 	return pubSub, nil
 }
+
+// Close cancels the context to stop the GCP receiver,
+// marks all unread messages from the channel the client did not receive them,
+// waits the GCP receiver will stop and releases the GCP Pub/Sub client.
+func (ps *PubSub) Close() error {
+	if ps == nil {
+		return nil
+	}
+
+	ps.ctxCancel()
+
+	for i := 0; i < pubsub.DefaultReceiveSettings.MaxOutstandingMessages; i++ {
```

Comment:
Maybe it would be better, if we close the `MessagesCh` [right after the Receive](https://github.com/conduitio-labs/conduit-connector-gcp-pubsub/pull/6/files#diff-4f10fdde6721d724dc515a4d7209740da160f27bacdb776db051f43aef87f37aR67), and here we'll just do the following:

```go
for msg := range ps.MessagesCh {
    msg.Nack()
}
```